### PR TITLE
STENCIL-2504 Show account creation only if it is enabled in store settings.

### PR DESCRIPTION
**Pull request by junedkazi**
This PR fixes 2 things;

- Sign In link is displayed always.
- Show the Sign Up link when the settings is enabled in the store settings.

@mcampa @mjschock @bigcommerce/stencil-team 

@nickdengler can we test this one.

### CODE REVIEW

**Comment by mattolson** [comment]
👍

**Comment by mcampa** [comment]
We are now using `CHANGELOG.md`. Please add an entry with your change under `draft` before merging

```
## Draft
- <here>

## 1.5.1 (2017-02-07)
- ...

```

**Comment by nickdengler** [comment]
If account creation is turned off, when the 'Sign in' link is clicked, you are taken to a blank page. Is there a BCApp pr that goes with this one?

**Comment by mcampa** [comment]
image

File: templates/pages/auth/create-account.html [hunk 1]
```diff
@@ -20,27 +20,26 @@ <h1 class="page-heading">{{lang 'create_account.heading' }}</h1>
...
-                                src="https://www.google.com/recaptcha/api/challenge?k={{forms.create_account.recaptcha.public_key}}">
-                        </script>
-                        <noscript>
-                            <iframe src="https://www.google.com/recaptcha/api/noscript?k={{forms.create_account.recaptcha.public_key}}" height="300" width="500" frameborder="0"></iframe>
-                            <br>
-                            <textarea name="recaptcha_challenge_field" rows="3" cols="40">
-                            </textarea>
-                            <input type="hidden" name="recaptcha_response_field" value="manual_challenge">
-                        </noscript>
+                    <div class="panel">
```
Comment (junedkazi) [inline templates/pages/auth/create-account.html]: Just indenting the markup.

File: templates/pages/auth/create-account.html [hunk 2]
```diff
@@ -20,27 +20,26 @@ <h1 class="page-heading">{{lang 'create_account.heading' }}</h1>
...
+                                </textarea>
+                                <input type="hidden" name="recaptcha_response_field" value="manual_challenge">
+                            </noscript>
+                        </div>
                     </div>
-                </div>
                 {{/if}}
                 <div class="form-actions">
-                    <input type="submit" class="button button--primary" value="{{lang 'forms.create_account.submit_value' }}" />
+                    <input type="submit" class="button button--primary" value="{{lang 'forms.create_account.submit_value' }}"  {{#unless settings.account_creation_enabled}}disabled{{/unless}} />
```
Comment (junedkazi) [inline templates/pages/auth/create-account.html]: Disable the submit button if account creation is disabled.

Comment (mcampa) [inline templates/pages/auth/create-account.html]: why not disable the entire form?
People will think is a bug if you only disable the the submit button

**Comment by junedkazi** [comment]
stencil-account-creation-disabled

**Comment by mcampa** [comment]
IMO we should not render the entire form if create account is disabled

**Comment by junedkazi** [comment]
I like the idea. 
stencil-do-not-render-form

**Comment by nickdengler** [comment]
2 small things:
- I don't see the error like you have it, instead it's just a blank page:
screen shot 2017-02-08 at 9 55 03 am

- The Sign In frame should be centered when the create account section is hidden:
screen shot 2017-02-08 at 9 54 48 am

**Comment by junedkazi** [comment]
@nickdengler to test 1 you will need this bcapp branch https://github.com/bigcommerce/bigcommerce/pull/17364

**Comment by nickdengler** [comment]
Ok got it now, LGTM 💚 

I also added a ticket for the centering (STENCIL-3011)

**Pull request merged**